frappe/tests/test_db.py

```diff
@@ -80,3 +86,57 @@ def test_log_touched_tables(self):
 		self.assertIn('tabCustom Field', frappe.flags.touched_tables)
 		frappe.flags.in_migrate = False
 		frappe.flags.touched_tables.clear()
+
+
+	def test_db_keywords_as_fields(self):
+		"""Tests if DB keywords work as docfield names. If they're wrapped with grave accents."""
+		# Using random.choices, picked out a list of 40 keywords for testing
+		all_keywords = {
+			"mariadb": ["CHARACTER", "DELAYED", "LINES", "EXISTS", "YEAR_MONTH", "LOCALTIME", "BOTH", "MEDIUMINT",
+			"LEFT", "BINARY", "DEFAULT", "KILL", "WRITE", "SQL_SMALL_RESULT", "CURRENT_TIME", "CROSS", "INHERITS",
+			"SELECT", "TABLE", "ALTER", "CURRENT_TIMESTAMP", "XOR", "CASE", "ALL", "WHERE", "INT", "TO", "SOME",
+			"DAY_MINUTE", "ERRORS", "OPTIMIZE", "REPLACE", "HIGH_PRIORITY", "VARBINARY", "HELP", "IS",
+			"CHAR", "DESCRIBE", "KEY"],
+			"postgres": ["WORK", "LANCOMPILER", "REAL", "HAVING", "REPEATABLE", "DATA", "USING", "BIT", "DEALLOCATE",
+			"SERIALIZABLE", "CURSOR", "INHERITS", "ARRAY", "TRUE", "IGNORE", "PARAMETER_MODE", "ROW", "CHECKPOINT",
+			"SHOW", "BY", "SIZE", "SCALE", "UNENCRYPTED", "WITH", "AND", "CONVERT", "FIRST", "SCOPE", "WRITE", "INTERVAL",
+			"CHARACTER_SET_SCHEMA", "ADD", "SCROLL", "NULL", "WHEN", "TRANSACTION_ACTIVE",
+			"INT", "FORTRAN", "STABLE"]
+		}
+		created_docs = []
+		fields = all_keywords[frappe.conf.db_type]
+		test_doctype = "ToDo"
+
+		def add_custom_field(field):
+			create_custom_field(test_doctype, {
+				"fieldname": field.lower(),
+				"label": field.title(),
+				"fieldtype": 'Data',
+			})
+
+		# Create custom fields for test_doctype
+		for field in fields:
+			add_custom_field(field)
+
+		# Create documents under that doctype and query them via ORM
+		for _ in range(10):
+			docfields = { key.lower(): random_string(10) for key in fields }
+			doc = frappe.get_doc({"doctype": test_doctype, "description": random_string(20), **docfields})
+			doc.insert()
+			created_docs.append(doc.name)
+
+		random_field = choice(fields).lower()
+		random_doc = choice(created_docs)
+		random_value = random_string(20)
+
+		# Testing read
+		self.assertEqual(list(frappe.get_all("ToDo", fields=[random_field], limit=1)[0])[0], random_field)
```

Comment:
Also, test one aggregate read, for e.g., `count(name) as total`

Reply:
That functionality isn't supported yet. Although I've added a test for aliases